docs/content/0.1.0/docs/deploying-dj/notifications.md

```diff
@@ -0,0 +1,93 @@
+---
+weight: 60
+title: Notifications
+---
+
+In DataJunction, notifications play a vital role in keeping users informed about various activity, such as changes in
+the availability state of nodes. This section discusses how notifications are managed within DataJunction and how you
+can implement a custom notification solution using FastAPI's dependency injection.
+
+### How Notifications are Used
+
+DataJunction uses notifications to alert users about significant events, such as updates to nodes or changes in their
+availability state. By handling notifications within the OSS project, DataJunction provides an opinionated take on where
+and when notifications should be sent.
+
+### Default Notification Implementation
+
+Out of the box, DataJunction includes a simple placeholder dependecny that simply logs notifications.
+
+Here's a brief look at this placeholder notification implementation:
+
+```py
+import logging
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+_logger = logging.getLogger(__name__)
+
+@dataclass
+class Message:
+    """A message for use in a notification"""
+
+    recipients: list[str]
+    subject: str
+    body: str
+    metadata: Optional[Dict[str, str]] = None
+
+def get_notifier():
+    """Returns a method for processing notifications"""
+
+    def notify(message: Message):
+        """Notify the recipients"""
+        _logger.debug("Sending notification to %s", message.recipients)
+        _logger.debug("Subject: %s", message.subject)
+        _logger.debug("Body: %s", message.body)
+        _logger.debug("Metadata: %s", message.metadata)
+
+    return notify
+```
+
+### Custom Notification Implementation
+
+You can implement a custom notification system by using FastAPI's dependency injection and injecting a `get_notifier`
+dependency. The custom notifier must handle the `notify` method, which processes the notification messages.
+
+#### Implementing a Custom Notifier
+
+To implement a custom notifier, create a function that processes notification messages and use FastAPI's dependency
+injection to inject your custom notifier.
+
+Here's an example of a custom notifier implementation:
```

Comment:
This may be silly, and not needed for seasoned engineers but I asking for a friend... :) Do you mind adding a note/sentence on where this change should happen?

Reply:
Great point, let me add the line for actually overriding the dependency in the app!

Reply:
Done